app/http/ResponseHeaders.scala

```diff
@@ -59,3 +59,26 @@ trait ResponseHeaders extends HeaderNames:
   def asAttachmentStream(name: String)(res: Result) = noProxyBuffer(asAttachment(name)(res))
 
   def lastModified(date: Instant) = LAST_MODIFIED -> date.atZone(utcZone)
+
+  object embedderPolicy:
+
+    def isSet(result: Result) = result.header.headers.contains(embedderPolicyHeader)
+
+    def forReq(req: RequestHeader) =
+      if supportsCoepCredentialless(req) then credentialless else requireCorp
+
+    def supportsCoepCredentialless(req: RequestHeader) =
+      import HTTPRequest.*
+      isChrome96Plus(req) || (isFirefox119Plus(req) && !isMobileBrowser(req))
+
+    def default        = headers("unsafe-none")
+    def credentialless = headers("credentialless")
+    def requireCorp    = headers("require-corp")
+
+    private val openerPolicyHeader   = "Cross-Origin-Opener-Policy"
+    private val embedderPolicyHeader = "Cross-Origin-Embedder-Policy"
+
+    private def headers(policy: "credentialless" | "require-corp" | "unsafe-none") = List(
+      openerPolicyHeader   -> "same-origin",
```

Comment:
```suggestion
      openerPolicyHeader -> if policy == "unsafe-none" then policy else "same-origin",
```

Reply:
oops, I shouldn't have merged that, it prevents setting other policies than these 2